llvm/lib/Target/RISCV/RISCVISelDAGToDAG.cpp

```diff
@@ -2441,6 +2441,22 @@ bool RISCVDAGToDAGISel::SelectAddrRegImm(SDValue Addr, SDValue &Base,
   return true;
 }
 
+bool RISCVDAGToDAGISel::SelectAddrRegImmLsb00000(SDValue Addr, SDValue &Base,
+                                                 SDValue &Offset) {
+
+  bool Found = SelectAddrRegImm(Addr, Base, Offset);
```

Comment:
I'm not sure we should use SelectAddrRegImm here. It creates new nodes for some cases, but then we might decide not to use them. I think we should directly handle the cases we want to handle.